service/cachemdw/middleware.go

```diff
@@ -0,0 +1,109 @@
+package cachemdw
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/kava-labs/kava-proxy-service/decode"
+)
+
+type cacheContextKey string
+
+const (
+	CachedContextKey cacheContextKey = "X-KAVA-PROXY-CACHED"
+
+	CacheHeaderKey       = "X-Cache"
+	CacheHitHeaderValue  = "HIT"
+	CacheMissHeaderValue = "MISS"
+)
+
+func (c *ServiceCache) Middleware(
+	next http.Handler,
+) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		uncachedContext := context.WithValue(r.Context(), CachedContextKey, false)
+
+		req := r.Context().Value(c.decodedRequestContextKey)
+		decodedReq, ok := (req).(*decode.EVMRPCRequestEnvelope)
+		if !ok {
+			next.ServeHTTP(w, r.WithContext(uncachedContext))
```

Comment:
here and anywhere else, don't have silent errors, log them including - what method failed, what data was provided to the function, and what the error was

Reply:
Added such log:
```go
c.Logger.Error().
	Str("method", r.Method).
	Str("url", r.URL.String()).
	Str("host", r.Host).
	Msg("can't cast request to *EVMRPCRequestEnvelope type")
```

lmk if I need to add request body in the log as well?
maybe use TeeReader to read body from http request and then log it?